### PR TITLE
chore: add lodash/fp

### LINKED ISSUE #18: Add `lodash/fp` to `babel-plugin-lodash` ids option

### PR DESCRIPTION
**Pull request by AfonsoVReis**
Closes #18 .

### CODE REVIEW

**Comment by andregoncalvesdev** [comment]
This is just an opinion but `defaultLodashOptionsIds` sounds overridable. I prefer `baseLodasOptionsIds` as it sounds more extensible like this case.

File: index.js [hunk 1]
```diff
@@ -94,9 +94,13 @@ module.exports = (context, options) => {
...
     if (options.lodash) {
+        const defaultLodashOptionsIds = ['lodash/fp'];
+
         config.plugins.push([
-            require.resolve('babel-plugin-lodash'),
-            options.lodash === true ? {} : options.lodash,
+            [require.resolve('babel-plugin-lodash')],
+            Object.assign({},
+                options.lodash,
+                { id: options.lodash.id ? defaultLodashOptionsIds.concat(options.lodash.id) : defaultLodashOptionsIds }),
```
Comment (satazor) [inline index.js]: We should document the defaults in the README and perhaps and example adding recompose?

**Comment by satazor** [comment]
I agree with having a more specific commit message.

File: index.js (hunk 1 above)
Comment (AfonsoVReis) [reply index.js]: I agree, i will add the information with an example in the README.

Comment (andregoncalvesdev) [reply index.js]: +1

File: index.js [hunk 2]
```diff
@@ -94,9 +94,13 @@ module.exports = (context, options) => {
 
     // Cherry-pick lodash modules for smaller bundles
     if (options.lodash) {
+        const defaultLodashOptionsIds = ['lodash/fp'];
```
Comment (satazor) [inline index.js]: Looking at https://github.com/lodash/babel-plugin-lodash/blob/master/src/config.js#L17, I think that when we pass `id`, it completely overrides the default ones. This means that we must include all those defaults.

Comment (AfonsoVReis) [reply index.js]: It has been reported in this issue. https://github.com/lodash/babel-plugin-lodash/issues/184 .
We can re-include the defaults, but we should be concerned about future fixes on this matter.

File: index.js [hunk 3]
```diff
@@ -94,9 +94,19 @@ module.exports = (context, options) => {
...
+            'lodash/fp',
+        ];
+
         config.plugins.push([
-            require.resolve('babel-plugin-lodash'),
-            options.lodash === true ? {} : options.lodash,
+            [require.resolve('babel-plugin-lodash')],
+            Object.assign({},
+                options.lodash,
+                { id: options.lodash.id ? baseLodashOptionsIds.concat(options.lodash.id) : baseLodashOptionsIds }),
```
Comment (satazor) [inline index.js]: Can this be changed to ?

```js
{ id: baseLodashOptionsIds.concat(options.lodash.id || []) }
```

It's a bit simpler.

File: test/__snapshots__/index.spec.js.snap [hunk 4]
```diff
@@ -43,9 +43,17 @@ Object {
...
+        "<PROJECT_ROOT>/node_modules/babel-plugin-lodash/lib/index.js",
+      ],
       Object {
-        "id": "lodash-compat",
+        "id": Array [
+          "lodash",
+          "lodash-es",
+          "lodash-compat",
+          "lodash/fp",
+          "lodash-compat",
```
Comment (satazor) [inline test/__snapshots__/index.spec.js.snap]: There's a repeated `lodash-compat` here. Can you change the test to use `recompose` instead?

File: README.md [hunk 5]
```diff
@@ -101,6 +101,21 @@ When `browsers` is specified, the compiled code will work on browsers that are s
 
 If you are developing a library or application that has different requirements in terms of browser or node support, you may specify the [targets](https://www.npmjs.com/package/babel-preset-env#targets) yourself as an object.
 
+### `lodash` option
+
+Transform to cherry-pick Lodash modules is enabled. 
+
+`lodash-es` `lodash-compat` and `lodash/fp` are added by default, but you can cherry-pick any module you need as long as you specify it on `options.id`.
```
Comment (satazor) [inline README.md]: Specify which modules will have the cherry-pick transformation applied.

Note that `lodash-es`, `lodash-compat` and `lodash/fp` are always added for you, regardless of having this option defined or not.

For instance, to have smaller bundles when using [recompose](https://github.com/acdlite/recompose):

File: index.js (hunk 3 above)
Comment (andregoncalvesdev) [reply index.js]: 👍

File: README.md [hunk 6]
```diff
@@ -101,6 +101,21 @@ When `browsers` is specified, the compiled code will work on browsers that are s
 
 If you are developing a library or application that has different requirements in terms of browser or node support, you may specify the [targets](https://www.npmjs.com/package/babel-preset-env#targets) yourself as an object.
 
+### `lodash` option
+
+Specify which modules will have the cherry-pick transformation applied.
+
+Note that `lodash-es`, `lodash-compat` and `lodash/fp` are always added for you, regardless of having this option defined or not.
+
+For instance, to have smaller bundles when using [recompose](https://github.com/acdlite/recompose) :
```
Comment (satazor) [inline README.md]: Please remove extra space before `:`.

**Comment by satazor** [comment]
@AfonsoVReis I would consider this a `feat` don't you think @andregoncalvesdev ?

**Comment by andregoncalvesdev** [comment]
Yes @satazor , i agree this is `feat`.

**Comment by AfonsoVReis** [comment]
Changed to feat instead of chore.

**Pull request merged**